range_queries/heavy_light_decomposition.cpp

```diff
@@ -69,514 +74,513 @@ namespace heavy_light_decomposition {
  * lift :- lift a node k units up the tree
  * kth_ancestor :- returns the kth ancestor
  * lca :- returns the least common ancestor
-*/
-template<typename X>
-class Tree {
-	// 
-	
+ */
+template <typename X> class Tree {
+  //
+
 private:
-	std::vector<std::list<int>> t_adj; ///> an adjacency list to stores the tree edges
-	const int t_nodes, t_maxlift; ///> number of nodes, and maximum possible height of the tree
-	std::vector<std::vector<int>> t_par; ///> a matrix to store every node's 2^kth parent
-	std::vector<int> t_depth, t_size; ///> a vector to store the depth of a node, and the subtree size rooted at node
-	int t_root; ///> the root of the tree
-	std::vector<X> t_val; ///> values of nodes
-	template<typename T> friend class HLD;
-	
-	/**
-	 * @brief Utility function to compute sub-tree sizes
-	 * @param u current dfs node
-	 * @param p the parent of node @param u
-	 * @returns void
-	*/
-	void dfs_size(int u, int p = -1) {
-		for(const int& v : t_adj[u]) {
-			if(v^p) {
-				dfs_size(v, u);
-				t_size[u] += t_size[v];
-			}
-		}
-	}
-
-	/**
-	 * @brief Utility function to populate the t_par vector
-	 * @param u current dfs node
-	 * @param p the parent of node u
-	 * @returns void
-	*/
-	void dfs_lca(int u, int p = -1) {
-		t_par[u][0] = p;
-		if(p != -1) t_depth[u] = 1 + t_depth[p];
-		for(int k = 1; k < t_maxlift; k++) {
-			if(t_par[u][k-1] != -1) {
-				t_par[u][k] = t_par[t_par[u][k-1]][k-1];
-			}
-		}
-
-		for(const int& v: t_adj[u]) {
-			if(v^p) {
-				dfs_lca(v, u);
-			}
-		}
-	}
+  std::vector<std::list<int>>
+      t_adj; ///> an adjacency list to stores the tree edges
```

Comment:
```suggestion
      t_adj; ///< an adjacency list to stores the tree edges
```
The comment is incorrect please note that the correct start tag is `///<` and not `///>`. 
Please make the corrections throughout the code.

Reply:
Looks like `clang-format` is breaking up the lines :confused:.